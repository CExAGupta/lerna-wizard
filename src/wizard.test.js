const {prompt, __setAnswers} = require('inquirer');
const wizard = require('./wizard');
afterEach(function () {
    __setAnswers([]);
});

[{
    testName: 'lerna ls',
    answers: [{command: 'ls'}, {logLevel: 'info'}]
}, {
    testName: 'custom log level',
    answers: [{command: 'ls'}, {logLevel: 'silly'}]
}, {
    testName: 'clean',
    answers: [{command: 'clean'}, {logLevel: 'info'}]
}, {
    testName: 'bootstrap',
    answers: [{command: 'bootstrap'}, {packages: ''}, {logLevel: 'info'}]
}, {
    testName: 'bootstrap but ignore some packages',
    answers: [{command: 'bootstrap'}, {packages: 'component-*'}, {logLevel: 'info'}]
}, {
    testName: 'publish',
    answers: [{command: 'publish'}, {shouldCustomize: false}, {logLevel: 'info'}]
}, {
    testName: 'publish custom: confirm all, no inputs',
    answers: [
        {command: 'publish'},
        {shouldCustomize: true},
        {canary: true},
        {skipGit: true},
        {skipNpm: true},
        {yes: true},
        {npmTag: ''},
        {forcePublishPackages: ''},
        {repoVersion: ''},
        {logLevel: 'info'}
    ]
}, {
    testName: 'publish custom: only inputs',
    answers: [
        {command: 'publish'},
        {shouldCustomize: true},
        {canary: false},
        {skipGit: false},
        {skipNpm: false},
        {yes: false},
        {npmTag: 'custom'},
        {forcePublishPackages: 'pkg1,pkg2'},
        {repoVersion: '1.0.0'},
        {logLevel: 'info'}
    ]
}, {
    testName: 'publish custom: no inputs, no confirms',
    answers: [
        {command: 'publish'},
        {shouldCustomize: true},
        {canary: false},
        {skipGit: false},
        {skipNpm: false},
        {yes: false},
        {npmTag: ''},
        {forcePublishPackages: ''},
        {repoVersion: ''},
        {logLevel: 'info'}
    ]
}, {
    testName: 'updated',
    answers: [{command: 'updated'}, {logLevel: 'info'}]
}, {
    testName: 'import',
    answers: [{command: 'import'}, {externalRepoPath: 'path/to/package'}, {logLevel: 'info'}]
}, {
    testName: 'diff all',
    answers: [{command: 'diff'}, {packageName: 'blank'}, {logLevel: 'info'}]
}, {
    testName: 'diff package',
    answers: [{command: 'diff'}, {packageName: 'pkg1'}, {logLevel: 'info'}]
}, {
    testName: 'init',
    answers: [{command: 'init'}, {independentMode: false}, {logLevel: 'info'}]
}, {
    testName: 'init',
    answers: [{command: 'init'}, {independentMode: true}, {logLevel: 'info'}]
}, {
    testName: 'run',
    answers: [{command: 'run'}, {scriptName: 'test'}, {concurrency: '4'}, {scope: ''}, {logLevel: 'info'}]
}, {
    testName: 'run, custom inputs',
    answers: [{command: 'run'}, {scriptName: 'test'}, {concurrency: '1'}, {scope: 'component-*'}, {logLevel: 'info'}]
}, {
    testName: 'run, empty inputs',
    answers: [{command: 'run'}, {scriptName: 'test'}, {concurrency: 0}, {scope: ''}, {logLevel: 'info'}]
}, {
    testName: 'exec',
    answers: [{command: 'exec'}, {commandName: 'npm version'}, {concurrency: '4'}, {scope: ''}, {logLevel: 'info'}]
}, {
    testName: 'exec, custom inputs',
    answers: [{command: 'exec'}, {commandName: 'npm version'}, {concurrency: '1'}, {scope: 'component-*'}, {logLevel: 'info'}]
}, {
    testName: 'exec, empty inputs',
    answers: [{command: 'exec'}, {commandName: 'npm version'}, {concurrency: 0}, {scope: ''}, {logLevel: 'info'}]
}].forEach(({testName, answers}) => {
    test(testName, async () => {
        __setAnswers(answers);
        return wizard().then(result => {
            expect(result).toMatchSnapshot();
        });
    })
});