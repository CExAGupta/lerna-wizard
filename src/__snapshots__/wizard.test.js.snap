// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bootstrap 1`] = `"lerna bootstrap"`;

exports[`bootstrap but ignore some packages 1`] = `"lerna bootstrap --ignore component-*"`;

exports[`clean 1`] = `"lerna clean"`;

exports[`custom log level 1`] = `"lerna --loglevel silly ls"`;

exports[`diff all 1`] = `"lerna diff"`;

exports[`diff package 1`] = `"lerna diff pkg1"`;

exports[`exec 1`] = `"lerna exec --concurrency 4 -- npm version"`;

exports[`exec, custom inputs 1`] = `"lerna exec --concurrency 1 --scope component-* -- npm version"`;

exports[`exec, empty inputs 1`] = `"lerna exec -- npm version"`;

exports[`import 1`] = `"lerna import path/to/package"`;

exports[`init 1`] = `"lerna init"`;

exports[`init 2`] = `"lerna init --independent"`;

exports[`lerna ls 1`] = `"lerna ls"`;

exports[`publish 1`] = `"lerna publish"`;

exports[`publish custom: confirm all, no inputs 1`] = `"lerna publish --canary --skip-git --skip-npm --yes"`;

exports[`publish custom: no inputs, no confirms 1`] = `"lerna publish"`;

exports[`publish custom: only inputs 1`] = `"lerna publish --npm-tag custom --force-publish pkg1,pkg2 --repo-version 1.0.0"`;

exports[`run 1`] = `"lerna run --concurrency 4 test"`;

exports[`run, custom inputs 1`] = `"lerna run --concurrency 1 --scope component-* test"`;

exports[`run, empty inputs 1`] = `"lerna run test"`;

exports[`updated 1`] = `"lerna updated"`;
